### PR TITLE
#277 initial helm chart implementation

### LINKED ISSUE #277: Helm chart for Kubernetes installation
# Use case

Kubernetes installation.

# Your idea for a solution

I have made a chart for my usage and it may also be fine for others at least fine to begin with.

- [x] I'm willing to help with the implementation

### PR DESCRIPTION
**Pull request by acim**

### CODE REVIEW

File: contrib/kubernetes/chart/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+            - name: data
+              mountPath: {{ .Values.persistence.path }}
+              {{- if .Values.persistence.subPath }}
+              subPath: {{ .Values.persistence.subPath }}
+              {{- end }}
+            - name: tls
+              mountPath: /etc/maddy/certs/mx.ectobit.com/fullchain.pem
+              subPath: tls.crt
+            - name: tls
+              mountPath: /etc/maddy/certs/mx.ectobit.com/privkey.pem
```
Comment (foxcpp) [inline contrib/kubernetes/chart/templates/deployment.yaml]: Is this domain a leftover from your experiments?

Comment (acim) [reply contrib/kubernetes/chart/templates/deployment.yaml]: Yes, leftovers, I forgot to rename this.

**Comment by acim** [comment]
Just tested locally using [kind](https://github.com/kubernetes-sigs/kind):

```
kubectl create secret tls maddy --cert=fullchain.pem --key=privkey.pem
helm upgrade --install maddy ./chart
```

and it runs happily:

```
smtp: listening on tcp://0.0.0.0:25	
sign_dkim: generating a new rsa2048 keypair...	
sign_dkim: generated a new rsa2048 keypair, private key is in dkim_keys/example.org_default.key, TXT record with public key is in dkim_keys/example.org_default.dns,
put its contents into TXT record for default._domainkey.example.org to make signing and verification work	
submission: listening on tls://0.0.0.0:465	
submission: listening on tcp://0.0.0.0:587	
imap: listening on tls://0.0.0.0:993	
imap: listening on tcp://0.0.0.0:143
```

**Comment by foxcpp** [comment]
Okay, I have no way of actually testing it so I will take your word for it. :)

**Comment by foxcpp** [comment]
Eeh, switching base branch messed it up. I will go merge it manually, I suppose...

**Comment by foxcpp** [comment]
Merged manually as 2b0fecbd3a00266e22517fa90429e88d47e66032.

**Pull request closed without merging**